src/org/rascalmpl/repl/rascal/RascalInterpreterREPL.java

```diff
@@ -132,14 +142,40 @@ public IDEServices initialize(Reader input, PrintWriter stdout, PrintWriter stde
             throw new IllegalStateException("Already initialized");
         }
         eval = buildEvaluator(input, stdout, stderr, services);
+
+        // Register watches for all watchable locations on the search path for automatic reloading
+        eval.getRascalResolver().collect().stream().filter(this::isWatchable).forEach(p -> {
+            try {
+                reg.watch(p, true, d -> sourceLocationChanged(p, d));
+            }
+            catch (IOException e) {
+                e.printStackTrace(stderr);
+            }
+        });
         return services;
     }
 
+    private boolean isWatchable(ISourceLocation loc) {
+        for (var capability : reg.capabilities(loc)) {
```

Comment:
I would prefer we add java functions to the registry so that we can ask this of a location. So that we get compile-time checking of the interface.

Reply:
Agreed